packages/monaco/src/browser/monaco-editor.ts

```diff
@@ -57,6 +60,13 @@ export class MonacoEditor implements TextEditor, IEditorReference {
         this.minHeight = options && options.minHeight !== undefined ? options.minHeight : -1;
         this.toDispose.push(this.create(options, override));
         this.addHandlers(this.editor);
+        this.registerDecorationTypes();
+    }
+
+    protected registerDecorationTypes(): void {
+        const decoarationTypes = this.decorationsService.getDecorationTypes();
+        const codeEditorService = this.editor._codeEditorService;
+        decoarationTypes.forEach(decoarationType => codeEditorService.registerDecorationType(decoarationType.type, decoarationType));
```

Comment:
what if decoration types have been changed? should we register new?

Reply:
well, it depends on, do we need it to be dynamic at all.

Reply:
I am not sure when extensions will register them, if they are assumptions about timing please document it in our API

Reply:
referring to https://github.com/theia-ide/theia/pull/1231#pullrequestreview-95126127, I want to change it, to be a contribution. what to you think?